src/main/java/org/terasology/gookeeper/component/MatingComponent.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.gookeeper.component;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityRef;
+
+public class MatingComponent implements Component {
```

Comment:
Akin to the hunger system this also already exists elsewhere (WildAnimals / WildAnimalsGenome). Any potential for reuse or are the systems expected to be too different? Could some base functionality from both systems be split out into a shared library module?

It doesn't have to contain the whole mating process, but maybe some parts like supporting conditions for when mating is available, logic for bringing partners together, and an event to trigger when the occasion has been consummated.

Not necessarily a thing for this PR, but it could be turned into a card for the backlog :-)